### PR TITLE
[User_accounts] Reject users cleanup 

### PR DESCRIPTION
**Pull request by zaliqarosli**
replacing #3548 because i somehow deleted the branch and closed the PR

moved PR to major branch

"This PR cleans up the code merged in #2811 as requested in comments by @driusan at the bottom of the page. This is a new PR to replace #3035 which is now closed because it was too messy.

Functionality should remain the same."

to test:
1. "Request Account" on login page
2. Check user in `users` table with Active='Y', Pending='Y'
2. Reject user on admin account via "Reject User" button in "User Accounts" module
3. Check user deleted from `users` table and there are no errors

### CODE REVIEW

**Comment by johnsaigle** [comment]
@zaliqarosli does this still `Need Work` or should we review/test?

**Comment by zaliqarosli** [comment]
@johnsaigle still needs work - i need to add a heck for when the user is not an instance of the AnonymousUser class

File: modules/user_accounts/ajax/rejectUser.php [hunk 1]
```diff
@@ -42,7 +42,7 @@
 $rejectee = \User::factory($_POST['identifier']);
 $username = $rejectee->getUsername();
 
-if (empty($username)) {
+if (empty($username) || $rejectee instanceof AnonymousUser) {
```
Comment (zaliqarosli) [inline modules/user_accounts/ajax/rejectUser.php]: @driusan is this what you meant?

**Comment by zaliqarosli** [comment]
@johnsaigle ready for review

**Comment by johnsaigle** [comment]
@zaliqarosli The original PR was issued against `minor` (which feels more appropriate). Was there a reason that this was changed to `major`?

**Comment by zaliqarosli** [comment]
@johnsaigle addition of the AnonymousUser class was made on major not minor. and changes made in this PR is big enough to deserve being in major as advised by @driusan

File: modules/user_accounts/ajax/rejectUser.php [hunk 2]
```diff
@@ -1,122 +1,63 @@
...
-    }
+if (!Edit_User::canRejectAccount($rejectee)) {
+    header("HTTP/1.1 403 Forbidden");
+    header("Content-Type: text/plain");
+    exit(
+        "This account is active and cannot be rejected"
+    );
+}
 
+(\Database::singleton())->delete('users', array("UserID" => $username));
```
Comment (ZainVirani) [inline modules/user_accounts/ajax/rejectUser.php]: can a not-yet-accepted user have any entries in `user_perm_rel`? if so, delete those too. if not, we good

Comment (zaliqarosli) [reply modules/user_accounts/ajax/rejectUser.php]: https://github.com/aces/Loris/blob/9cefbcd13844cb44bc4c979eeb9af962676bdbf6/SQL/0000-00-01-Permission.sql#L59

yes, a user pending approval can have entries in `user_perm_rel` but the FK on that table has an on delete cascade clause to it so permissions are automatically deleted along with the entry is `users`

File: modules/user_accounts/ajax/rejectUser.php [hunk 3]
```diff
@@ -1,122 +1,63 @@
...
-    // @codingStandardsIgnoreStart
-    function reject_user($identifier)
-    {
-    // @codingStandardsIgnoreEnd
-        $DB      = Database::singleton();
-        $config  = NDB_Config::singleton();
-        $baseURL = $config->getSetting('url');
-
-        $redirect = $baseURL . "/user_accounts/";
+if (empty($username) || $rejectee instanceof AnonymousUser) {
```
Comment (zaliqarosli) [inline modules/user_accounts/ajax/rejectUser.php]: @driusan would i need \AnonymousUser here or is this fine?

**Pull request merged**